File: LC217.java
// Solution to Leetcode 217: Contains Duplicate
// Category: Arrays and Hashing
// Difficulty: Easy

import java.util.HashMap;
 
public class LC217 {
    public boolean containsDuplicate(int[] nums) {
        HashMap<Integer, Integer> map = new HashMap<>();
        for (int i = 0; i < nums.length; i++)
        {
            if (!map.containsKey(nums[i]))
            {
                map.put(nums[i],0);
            }
            else
            {
                return true;
            }
        }
        return false;
    }
}

File: Python/LC217.py
# LeetCode 217: Contains Duplicate
# Difficulty: Easy
# Arrays
# Time Complexity: O(n)

from ast import List


class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        elements = set()
        for num in nums:
            if num in elements:
                return True
            elements.add(num)
        return False